frontend/src/metabase/components/AppBanner/AppBanner.tsx

```diff
@@ -3,15 +3,19 @@ import { useSelector } from "metabase/lib/redux";
 import { PaymentBanner } from "metabase/nav/components/PaymentBanner/PaymentBanner";
 import { ReadOnlyBanner } from "metabase/nav/components/ReadOnlyBanner";
 import { getUserIsAdmin } from "metabase/selectors/user";
+import { getIsHosted } from "metabase/setup/selectors";
 
 export const AppBanner = () => {
   const isAdmin = useSelector(getUserIsAdmin);
+  const isHosted = useSelector(getIsHosted);
   const tokenStatus = useSetting("token-status");
   const readOnly = useSetting("read-only-mode");
 
   const paymentStatuses = ["past-due", "unpaid", "invalid"];
   const shouldRenderPaymentBanner =
-    tokenStatus && paymentStatuses.includes(tokenStatus?.status ?? "");
+    !isHosted &&
+    tokenStatus &&
+    paymentStatuses.includes(tokenStatus?.status ?? "");
```

Comment:
Every string includes an empty string, so if `tokenStatus.status` is `undefined` this will be `true`

Reply:
![image](https://github.com/user-attachments/assets/55b8a1c6-c777-4bdd-baee-3b70e67619f5)

Reply:
Added a test for this case as well. https://github.com/metabase/metabase/pull/50337/commits/06c2de326359a3845f73c012d0c79b08cac115cd

```
  AppBanner
    PaymentBanner
      ✓ should render past-due banner for admin user with tokenStatusStatus: past-due (22 ms)
      ✓ should render unpaid banner for admin user with tokenStatusStatus: unpaid (12 ms)
      ✓ should render an error with details when the token is `invalid` (19 ms)
      ✓ should not render for admin user with tokenStatusStatus: something-else (10 ms)
      ✓ should not render when token status is missing (13 ms) <<<<<<<<<
      ✓ should not render for hosted instances for past-due token status (metabase#50335) (11 ms)
      ✓ should not render for hosted instances for unpaid token status (metabase#50335) (10 ms)
      ✓ should not render for hosted instances for invalid token status (metabase#50335) (13 ms)
```

Reply:
> Every string includes an empty string

This is true for `String.prototype.includes` but we're operating on `Array.prototype.includes`.

Reply:
Ah, it's an array, all good :+1: